src/App.js

```diff
@@ -26,14 +26,14 @@ function App() {
   useEffect(() => {
     Promise.all(
       moviesToQuery.map((movie) => {
-        fetch(`http://www.omdbapi.com/?apikey=${API_KEY}&t=${movie.imdbID}`);
+        return fetch(`http://www.omdbapi.com/?apikey=${API_KEY}&t=${movie.imdbID}`);
       })
     )
       .then((values) => console.log("Values: ", values))
       .catch(function (error) {
         console.log("Requestfailed", error);
       });
-  }, []);
+  }, [moviesToQuery]);
```

Comment:
The deps array here is important. With just an empty array, this only runs on mount. But `moviesToQuery` is just an empty array on the initial mount. Its not set to the movies until after the fetch to search by movies resolves and then you set state with the result. By adding `moviesToQuery` to the dependency array, we ensure that this useEffect is triggered every time `moviesToQuery` changes